hvplot/plotting/scatter_matrix.py

```diff
@@ -38,19 +47,99 @@ def scatter_matrix(data, c=None, chart='scatter', diagonal='hist', alpha=0.5, **
     diagonal = HoloViewsConverter._kind_mapping[diagonal]
     chart = HoloViewsConverter._kind_mapping[chart]
 
+    if rasterize:
+        if dynspread or spread:
+            if hd.ds_version < '0.12.0':
+                raise RuntimeError('Any version of datashader ' +
+                                    'less than 0.12.0 does not support ' + 
+                                    'rasterize with dynspread or spread')
+    #remove datashade kwds
+    if datashade or rasterize:
+        ds_kwds = {}
+        if 'aggregator' in kwds:
+            ds_kwds['aggregator'] = kwds.pop('aggregator')
+        
+    #remove dynspread kwds
+    if dynspread:
+        if datashade == False and rasterize == False:
+            warnings.warn(
+                "Datashade or Rasterize must be specified to use dynspread. " 
+                "Dynspread will not be applied to plots."
+                )
+
+        dn_kwds = {}
+        if 'max_px' in kwds:
+            dn_kwds['max_px'] = kwds.pop('max_px')
+        if 'threshold' in kwds:
+            dn_kwds['threshold'] = kwds.pop('threshold')
+        if 'how' in kwds:
+            dn_kwds['how'] = kwds.pop('how')
+        if 'mask' in kwds:
+            dn_kwds['mask'] = kwds.pop('mask')
```

Comment:
```suggestion
        if 'max_px' in kwds:
            sp_kwds['max_px'] = kwds.pop('max_px')
        if 'threshold' in kwds:
            sp_kwds['threshold'] = kwds.pop('threshold')
        if 'how' in kwds:
            sp_kwds['how'] = kwds.pop('how')
        if 'mask' in kwds:
            sp_kwds['mask'] = kwds.pop('mask')
```

(needed for changes below; only either dynspread or spread can be used at one time, so we can pool their options.)